doc/stdenv/stdenv.chapter.md

```diff
@@ -1077,9 +1077,9 @@ A few libraries automatically add to `NIX_LDFLAGS` their library, making their s
 
 The `validatePkgConfig` hook validates all pkg-config (`.pc`) files in a package. This helps catching some common errors in pkg-config files, such as undefined variables.
 
-### cmake {#cmake}
+### CMake {#cmake}
 
-Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. Some additional flags are passed in to give similar behavior to configure-based packages. You can disable this hook’s behavior by setting configurePhase to a custom value, or by setting dontUseCmakeConfigure. cmakeFlags controls flags passed only to CMake. By default, parallel building is enabled as CMake supports parallel building almost everywhere. When Ninja is also in use, CMake will detect that and use the ninja generator.
+Adds dependencies to the CMAKE_PREFIX_PATH, CMAKE_LIBRARY_PATH, CMAKE_INCLUDE_PATH and CMAKE_FRAMEWORK_PATH environment variables (cannot be disabled). Overrides the `configurePhase` to run the CMake configure and generate step (disable by setting `dontUseCmakeConfigure`, or `configurePhase`). The `configurePhase` adds the content of `cmakeFlags` to the command run. The build, install and check phases are not modified. If `ninja` is a dependency, the Ninja setup hook applies. Otherwise, the Make-based phases are used. The `configurePhase` selects the adequate CMake generator for the build, install and check phases to work. As a result, to prevent a Ninja-based build, you need to disable all three phases, as described in the section below. Enables parallel building (disable by setting `enableParallelBuilding` to 0).
```

Comment:
To be honest, I find the previous order of facts better – the fact that it runs CMake (which includes configure and generate step) is IMO more important than the variables it sets (after all, the variables should really only be interesting to CMake), so it should be mentioned first.

Reply:
I wrote this PR when I tried to understand how nix was using CMake. The existing wording was not sufficient for me to understand which knobs nix was turning to drive CMake. It gave a basic overview but required digging into the code to get the full picture.

I had kept the format at a single paragraph as it was, but maybe we should have a paragraph with the high level information, and a bullet list of the actual changes from the generic `stdenv` and how each can be disabled. This information is helpful for people who want to use CMake to write a derivation and control the behavior. What do you think?

Reply:
Sounds good, maybe also look at https://github.com/NixOS/nixpkgs/pull/126032 for inspiration.